alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,99 @@
+use std::ffi::{CString, CStr};
+use std::io;
+use std::mem;
+use std::os::raw;
+use std::path::PathBuf;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+    let flavor = PROC_PIDVNODEPATHINFO as raw::c_int;
+    let mut info = mem::MaybeUninit::<proc_vnodepathinfo>::uninit();
+    let size = mem::size_of::<proc_vnodepathinfo>() as raw::c_int;
+    let c_str = unsafe {
+        let result = proc_pidinfo(pid, flavor, 0, info.as_mut_ptr() as *mut raw::c_void, size);
+        match result {
+            c if c <= 0    => return Err(io::Error::last_os_error()),
+            s if s != size => return Err(io::Error::new(io::ErrorKind::Other, "invalid return size")),
+            _ => {}
+        };
+        CStr::from_ptr(info.assume_init().pvi_cdir.vip_path.as_ptr())
+    };
+
+    match CString::from(c_str).into_string() {
+        Err(e) => Err(io::Error::new(io::ErrorKind::Other, e)),
+        Ok(s) => Ok(PathBuf::from(s)),
+    }
```

Comment:
```suggestion
    CString::from(c_str).into_string()
        .map(PathBuf::from)
        .map_err(|err| IOError::new(IOErrorKind::Other, err));
    }
```

I think both would be fine here, but I feel like since we just remap both it might be nicer to do in by mapping?